sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1239,6 +1254,24 @@ object SQLContext {
     instantiatedContext.compareAndSet(null, sqlContext)
```

Comment:
We use  `compareAndSet `, so the one created by `newSession` will NOT overwrite the original one.